lib/open_weather/request.rb

```diff
@@ -34,7 +34,11 @@ def request(method, path, options)
         when :post, :put
           request.path = path
           request.params = { appid: options.delete(:appid) }
-          request.body = options.to_json unless options.empty?
+          if options.key?(:body)
+            request.body = options.delete(:body).to_json
+          elsif options.present?
+            request.body = options.to_json
+          end
```

Comment:
This method is starting to look like it's doing too much >_< there's a lot of implicit logic in `options`

Reply:
I think it's a fine kitchen-sink method :)